packages/bigquery/src/index.js

```diff
@@ -76,6 +77,284 @@ function BigQuery(options) {
+ * @param {string|number} value.month - One or two digits.
+ * @param {string|number} value.day - One or two digits.
+ * @param {string=|number=} value.hours - One or two digits (`00` - `23`).
+ * @param {string=|number=} value.minutes - One or two digits (`00` - `59`).
+ * @param {string=|number=} value.seconds - One or two digits (`00` - `59`).
+ * @param {string=|number=} value.fractional - Up to six digits for microsecond
+ *     precision.
+ *
+ * @example
+ * var datetime = bigquery.datetime('2017-01-01');
+ *
+ * //-
+ * // Alternatively, provide an object.
+ * //-
+ * var datetime = bigquery.datetime({
+ *   year: 2017,
+ *   month: 1,
+ *   day: 1,
+ *   hours: 14,
+ *   minutes: 0,
+ *   seconds: 0
+ * });
+ */
+BigQuery.datetime =
+BigQuery.prototype.datetime = function(value) {
+  if (!(this instanceof BigQuery.datetime)) {
+    return new BigQuery.datetime(value);
+  }
+
+  if (is.object(value)) {
+    var time;
+
+    if (value.hours) {
+      time = BigQuery.time(value).value;
+    }
+
+    value = format('{y}-{m}-{d}{time}', {
+      y: value.year,
+      m: value.month,
+      d: value.day,
+      time: time ? ' ' + time : ''
+    });
+  }
+
+  this.value = value;
+};
+
+/**
+ * A `TIME` data type represents a time, independent of a specific date.
+ *
+ * @param {object|string} value - The time. If a string, this should be in the
+ *     format the API describes: `[H]H:[M]M:[S]S[.DDDDDD]`. Otherwise, provide
+ *     an object.
+ * @param {string=|number=} value.hours - One or two digits (`00` - `23`).
+ * @param {string=|number=} value.minutes - One or two digits (`00` - `59`).
+ * @param {string=|number=} value.seconds - One or two digits (`00` - `59`).
+ * @param {string=|number=} value.fractional - Up to six digits for microsecond
+ *     precision.
+ *
+ * @example
+ * var time = bigquery.time('14:00:00'); // 2:00 PM
+ *
+ * //-
+ * // Alternatively, provide an object.
+ * //-
+ * var time = bigquery.time({
+ *   hours: 14,
+ *   minutes: 0,
+ *   seconds: 0
+ * });
+ */
+BigQuery.time =
+BigQuery.prototype.time = function(value) {
+  if (!(this instanceof BigQuery.time)) {
+    return new BigQuery.time(value);
+  }
+
+  if (is.object(value)) {
+    value = format('{h}:{m}:{s}{f}', {
+      h: value.hours,
+      m: value.minutes || 0,
+      s: value.seconds || 0,
+      f: is.defined(value.fractional) ? '.' + value.fractional : ''
+    });
+  }
+
+  this.value = value;
+};
+
+/**
+ * A timestamp represents an absolute point in time, independent of any time
+ * zone or convention such as Daylight Savings Time.
+ *
+ * @param {date} value - The time.
+ *
+ * @example
+ * var timestamp = bigquery.timestamp(new Date());
+ */
+BigQuery.timestamp =
+BigQuery.prototype.timestamp = function(value) {
+  if (!(this instanceof BigQuery.timestamp)) {
+    return new BigQuery.timestamp(value);
+  }
+
+  value = value || new Date();
+
+  if (is.date(value)) {
+    value = value.toJSON().replace(/^(.*)T(.*)Z$/, '$1 $2');
```

Comment:
@stephenplusplus I feel this bigquery/issue was what I filed on your behalf :tada: , to request BigQuery server side to accept this universal RFC3339 timestamp format (from the Date.toISOString(), or toJSON), and I just got notified 2 days ago they made the change, hence since now on you may drop the .replace call because it should no longer be necessary?

https://code.google.com/p/google-bigquery/issues/detail?id=842#c6